argocd/provider_test.go

```diff
@@ -77,6 +77,6 @@ func testAccPreCheckFeatureNotSupported(t *testing.T, feature int) {
 		t.Fatal("feature constraint is not handled by the provider")
 	}
 	if i := versionConstraint.Compare(serverVersion); i != 1 {
-		t.Skipf("version %s does not support feature", v)
+		t.Skipf("not running test if feature is already supported (v%s)", v)
```

Comment:
Not sure I follow the logic here? Tests should be run if the feature is supported. They should be skipped when the feature is not supported.

Reply:
It is used in this test : https://github.com/oboukili/terraform-provider-argocd/pull/226/files#diff-6e254abd82d715b01ed5073e07acb0324eb3dc17fe7e13d9c0f6134b85cc26e3R110-R137 (making sure the "not supported" error is raised)
Since we are using a matrix during tests, there is no point to run this test if the version supports it (IMHO, the previous wording was misleading)
While writing this, I noticed the v2.2.0 is out of tested versions, so the test wasn't needed at all

Reply:
> IMHO, the previous wording was misleading

You are quite correct - my bad. Was mixing up this function, `testAccPreCheckFeature**Not**Supported`, with `testAccPreCheckFeatureSupported`.